### ISSUE TITLE
AVT1: Orphan content violation in Tooltip

### ISSUE THREAD

**Issue by xiepingp** [carbon-design-system/carbon#6308]
Moving from:  [alchemy-containers/argonauts-ui#1176](https://github.ibm.com/alchemy-containers/argonauts-ui/issues/1176)

=====ENVIRONMENT=====
Browser: Chrome
DAP tool (version is 1.8.0.0) with "IBM Accessibility January 2020" and "IBM Accessibility Experimental" supported rulesets.

=====STEPS TO REPRODUCE=====
Go to https://cloud.ibm.com/observe/ -> click "Create new" button (LogDNA or Sysdig or Activitytracker)
LogDNA: on LogDNA instance creation page, click tooltip icon besides 'Select a resource group' or besides 'Tags'
Sysdig: on Sysdig instance creation page, click tooltip icon besides 'Select a resource group' or besides 'Tags'
Activitytracker: on Activitytracker instance creation page, click tooltip icon besides 'Select a resource group' or besides 'Tags'

=====DEFECT DESCRIPTION=====
Issue: Orphan content found that does not reside within a WAI-ARIA landmark or labelled region role.

=====EXPECTED RESULT=====
https://aat.w3ibm.mybluemix.net/token/ffe6145a-21af-417b-b1c1-a34e72f4ac64/d9578cd4-659f-48de-a06f-1ad6eb41766c/archives/2020JanDeploy//doc/w3/help/en-US/idhi_accessibility_check_g1157.html

LogDNA:
![1176](https://user-images.githubusercontent.com/63960906/84974201-5e42e400-b155-11ea-99b9-eab38d749c61.png)

![catalog_services_ibm-log-analysis-with-logdna_tooltip](https://media.github.ibm.com/user/277747/files/3afa8a00-946f-11ea-96ce-a0ea90f4fdae)

Sysdig:
![catalog_services_ibm-cloud-monitoring-with-sysdig_tooltip](https://media.github.ibm.com/user/277747/files/43eb5b80-946f-11ea-8dde-f237b11d8f65)

Activitytracker:
![catalog_services_ibm-cloud-activity-tracker-with-logdna_tooltip](https://media.github.ibm.com/user/277747/files/48b00f80-946f-11ea-945b-13cc04ed8755)

**Comment by joshblack** [comment]
cc @dakahn how should we handle the portal/landmark violation from DAP? In the past we talked about this being a false positive, right?

**Comment by dakahn** [comment]
@joshblack let me do some testing and reproduce this error. if memory serves me right it wasn't _truly_ orphaned, yeah

**Comment by joshblack** [comment]
@dakahn got it, let me know what you find!

**Comment by dakahn** [comment]
Yeah, this seems like something breaking on DAP/AC's end. Opening a tooltip and inspecting the code the elements in question are very much descendants of the page's <body>
![image](https://user-images.githubusercontent.com/40970507/87484742-9aeee780-c5fc-11ea-9ec5-c59097f68e77.png)

I'll make a support issue for AC's engine and link it here. Leaving this open for now.

**Comment by dakahn** [comment]
So word back from HQ is we should maybe set `role="region"` on the tooltip dialog to get this error to go away. This does seem to be AC/DAP handling dialogs in a non-appy way, but I'm still waiting to hear more.

**Comment by dakahn** [comment]
Running latest DAP/Accessibility Checker doesn't throw this error anymore.

**Comment by xiepingp** [comment]
@dakahn , would you mind advising which carbon version fixes this issue?

**Comment by xiepingp** [comment]
@dakahn, our dev is wondering which carbon version having this fix so that he could do the update accordingly. Could you please advise?

**Comment by mashenka123** [comment]
@dakahn Is it the carbon 10.17.0 which has the fix? Please kindly confirm, thank you!

**Comment by xiepingp** [comment]
The Orphan content violation in Tooltip has gone, but new violation appears on the same element <span class="bx--visually-hidden" role="link" tabindex="0">
![1176 - retest](https://media.github.ibm.com/user/277747/files/5e19b980-f814-11ea-8469-8522a804da56)
![1176 - retest  - details](https://media.github.ibm.com/user/277747/files/5eb25000-f814-11ea-82bf-62e3fd189977)
Not sure it is a Carbon bug or not...
@flannanl fyi.

**Comment by flannanl** [comment]
Seems like the violation is from the `<span>` element in the tooltip. So I believe this is from carbon as well.

<img width="1678" alt="image" src="https://user-images.githubusercontent.com/12156672/93335243-226ef780-f7f4-11ea-98c4-7d883cebf8ad.png">

**Comment by dakahn** [comment]
Reopening!

**Comment by dakahn** [comment]
I'm unable to recreate this issue on our end
![image](https://user-images.githubusercontent.com/40970507/93383540-62040680-f829-11ea-8e36-bb7fa75b1fa8.png)

@flannanl @xiepingp 
Can you test and confirm you see this issue [at this link](https://react.carbondesignsystem.com/iframe.html?id=tooltip--default-bottom) so we can figure out if this is an AC bug, a bug in our implementation or yours.

**Comment by xiepingp** [comment]
I didn't see the violation in the storybook either, @flannanl may you have a look?
![Screen Shot 2020-09-17 at 10 06 06 AM](https://user-images.githubusercontent.com/63960906/93411309-74d01700-f8cd-11ea-94fe-6d2f9d45f81e.png)

**Comment by flannanl** [comment]
The complaining element is coming from Tooltip component for sure. When the tooltip is closed, the `span` element will be removed. I am also seeing the same `span` element in the storybook, but somehow it is not complained in the storybook. What I found is that the tooltip element in the storybook is created in an iframe where the one in the Console page is a direct child element under the `body`.

<img width="1680" alt="image" src="https://user-images.githubusercontent.com/12156672/93465926-6ed03b00-f8b9-11ea-9d8c-0c9b0e0c7b0a.png">

**Comment by flannanl** [comment]
And the carbon-react version we picked up is v7.18.

**Comment by dakahn** [comment]
I'm unable to recreate that on [our docs site](https://www.carbondesignsystem.com/components/tooltip/usage#top) as well

could you update to latest 7.20?

**Comment by flannanl** [comment]
@dakahn I cannot move up carbon. The cloudPAL team controls when to move up the carbon version.

**Comment by dakahn** [comment]
@flannanl let me know if you're able to recreate your bug on either linked page above. If not we can try and figure out what's happening in your code.

**Comment by flannanl** [comment]
@dakahn Yes, I'm still seeing it with 7.20

**Comment by flannanl** [comment]
<img width="1680" alt="image" src="https://user-images.githubusercontent.com/12156672/94469618-679c0d80-0194-11eb-8356-4b949f04d92b.png">

**Comment by emyarod** [comment]
I am unable to reproduce this error

![image](https://user-images.githubusercontent.com/8265238/112195035-7f020380-8bd7-11eb-9891-f02c564cc709.png)

![image](https://user-images.githubusercontent.com/8265238/112195105-92ad6a00-8bd7-11eb-9e74-b1334883b994.png)

**Comment by dakahn** [comment]
Confirmed. No longer can reproduce. They might have updated the ruleset on us. Closing as resolved 👍🏽
